### PR TITLE
DOCS-6238 Add Test Visibility and Intelligent Test Runner Documentation

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do? What is the motivation?

Extracts Test Visibility and ITR documentation from CI Visibility doc set along with guides and troubleshooting content. Cleans up legacy links.

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

File: content/en/intelligent_test_runner/troubleshooting/_index.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+<div class="alert alert-warning">Intelligent Test Runner is not available in the selected site ({{< region-param key="dd_site_name" >}}) at this time.</div>
+{{< /site-region >}}
+
+## Overview
+
+This page provides information to help you troubleshot issues with Intelligent Test Runner. If you need additional help, contact [Datadog Support][2].
+
+## Intelligent Test Runner is not working
+
+[Intelligent Test Runner][12] works by analyzing your commit history along with code coverage information about past test runs to determine which tests need to be run and which ones can be safely skipped. A minimum amount of information needs to exist in order for the Intelligent Test Runner to work correctly:
```
Comment (drichards-87) [inline content/en/intelligent_test_runner/troubleshooting/_index.md:20]:
```suggestion
[Intelligent Test Runner][12] works by analyzing your commit history along with code coverage information about past test runs to determine which tests need to be run and which ones can be safely skipped. A minimum amount of information needs to exist in order for Intelligent Test Runner to work correctly:
```

**Review by drichards-87 — CHANGES_REQUESTED** [review]
Looks good overall! I just had a few minor suggestions + I noted a couple of broken links.

File: content/en/tests/_index.md [hunk 2]
```diff
@@ -38,17 +39,17 @@ cascade:
...
-    {{< nextlink href="continuous_integration/tests/setup/swift" >}}Swift{{< /nextlink >}}
-    {{< nextlink href="continuous_integration/tests/junit_upload" >}}Uploading JUnit test report files to Datadog{{< /nextlink >}}
-    {{< nextlink href="continuous_integration/tests/containers" >}}Tests running in containers{{< /nextlink >}}
+    {{< nextlink href="tests/setup/dotnet" >}}.NET{{< /nextlink >}}
+    {{< nextlink href="tests/setup/java" >}}Java{{< /nextlink >}}
+    {{< nextlink href="tests/setup/javascript" >}}JavaScript{{< /nextlink >}}
+    {{< nextlink href="tests/setup/python" >}}Python{{< /nextlink >}}
+    {{< nextlink href="tests/setup/ruby" >}}Ruby{{< /nextlink >}}
+    {{< nextlink href="tests/setup/swift" >}}Swift{{< /nextlink >}}
+    {{< nextlink href="tests/junit_upload" >}}Uploading JUnit test report files to Datadog{{< /nextlink >}}
```
[the remark this answers is not in the document]
Comment (alai97) [reply content/en/tests/_index.md:48]:
```suggestion
    {{< nextlink href="tests/junit_xml" >}}Uploading JUnit test report files to Datadog{{< /nextlink >}}
```

File: content/en/tests/browser_tests.md [hunk 3]
```diff
@@ -52,6 +53,6 @@ The RUM session has all the data that [RUM normally collects][3] so you can debu
 
 {{< partial name="whats-next/whats-next.html" >}}
 
-[1]: /continuous_integration/tests/javascript/?tab=cypress#instrument-your-tests
+[1]: /tests/javascript/?tab=cypress#instrument-your-tests
```
[the remark this answers is not in the document]
Comment (alai97) [reply content/en/tests/browser_tests.md:56]:
```suggestion
[1]: /tests/setup/javascript/?tab=cypress#instrument-your-tests
```

File: content/en/tests/_index.md [hunk 4]
```diff
@@ -45,7 +45,7 @@ cascade:
     {{< nextlink href="tests/setup/python" >}}Python{{< /nextlink >}}
     {{< nextlink href="tests/setup/ruby" >}}Ruby{{< /nextlink >}}
     {{< nextlink href="tests/setup/swift" >}}Swift{{< /nextlink >}}
-    {{< nextlink href="tests/junit_upload" >}}Uploading JUnit test report files to Datadog{{< /nextlink >}}
+    {{< nextlink href="tests/junit_xml" >}}Uploading JUnit test report files to Datadog{{< /nextlink >}}
```
Comment (drichards-87) [inline content/en/tests/_index.md:48]:
```suggestion
    {{< nextlink href="tests/setup/junit_xml" >}}Uploading JUnit test report files to Datadog{{< /nextlink >}}
```

**Review by romainkomorndatadog — APPROVED** [review]
This looks good to me from the libraries side, but I'd like to defer to @KickingTheTV for the way this makes ITR a product separate from test visibility, so my approval's contingent on his approval.

**Review by KickingTheTV — APPROVED** [review]
LGTM

**Pull request merged**